lib/blueprinter/base.rb

```diff
@@ -49,18 +50,30 @@ def self.identifier(method, name: method, serializer: PublicSendSerializer)
     # @option options [Symbol] :name Use this to rename the method. Useful if
     #   if you want your JSON key named differently in the output than your
     #   object's field or method name.
+    # @yield [Object] The object passed to `render` is also passed to the
+    #   block.
     #
     # @example Specifying a user's first_name to be serialized.
     #   class UserBlueprint < Blueprinter::Base
     #     field :first_name
     #     # other code
     #   end
     #
+    # @example Passing a block to be evaluated as the value.
+    #   class UserBlueprint < Blueprinter::Base
+    #     field :full_name {|obj| "#{obj.first_name} #{obj.last_name}"}
+    #     # other code
+    #   end
+    #
     # @return [Field] A Field object
-    def self.field(method, options = {})
-      name = options.delete(:name) || method
-      serializer = options.delete(:serializer) || AssociationSerializer
-      current_view << Field.new(method, name, serializer, options)
+    def self.field(method, options = {}, &block)
+      options = if block_given?
+        {name: method, serializer: BlockSerializer, block: {method => block}}
+      else
+        {name: method, serializer: AssociationSerializer}
+      end.merge options
+      current_view << Field.new(method, options[:name], options[:serializer],
+        options)
```

Comment:
NB: The `options)` bugs me - would love to clean up the formatting of this line

Reply:
will do